e2e/playwright/support/init_test.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import path from 'node:path';
+import fs from 'node:fs';
+
+import {test} from '@e2e-support/test_fixture';
+import {DeepPartial} from '@mattermost/types/utilities';
+import {AdminConfig} from '@mattermost/types/config';
+
+import {cleanUpBotDMs} from './utils';
+import {clearKVStoreForPlugin} from './kv';
+import {preferencesForUser} from './user';
+
+const pluginDistPath = path.join(__dirname, '../../../dist');
+const pluginId = 'com.mattermost.plugin-todo';
+
+// # One time tasks
+test.beforeAll(async ({pw}) => {
+    const {adminClient, adminUser} = await pw.getAdminClient();
+    if (adminUser === null) {
+        throw new Error('can not get adminUser');
+    }
+
+    // Clear KV store
+    await clearKVStoreForPlugin(pluginId);
+
+    // Upload and enable plugin
+    const files = await fs.promises.readdir(pluginDistPath);
+    const bundle = files.find((fname) => fname.endsWith('.tar.gz'));
+    if (!bundle) {
+        throw new Error('Failed to find plugin bundle in dist folder');
+    }
+
+    const bundlePath = path.join(pluginDistPath, bundle);
+    await adminClient.uploadPluginX(bundlePath, true);
+    await adminClient.enablePlugin(pluginId);
+
+    // Configure plugin
+    const config = await adminClient.getConfig();
+    const newConfig: DeepPartial<AdminConfig> = {
+        ServiceSettings: {
+            EnableTutorial: false,
+            EnableOnboardingFlow: false,
+        },
+        PluginSettings: {
+            ...config.PluginSettings,
+            Plugins: {
+                ...config.PluginSettings.Plugins,
+                [pluginId]: todoConfig as any,
+            },
+        },
+    };
+
+    await adminClient.patchConfig(newConfig);
+    await adminClient.savePreferences(adminUser.id, preferencesForUser(adminUser.id));
+});
+
+// # Clear bot DM channel
+test.beforeEach(async ({pw}) => {
+    const {adminClient, adminUser} = await pw.getAdminClient();
+    if (adminUser === null) {
+        throw new Error('can not get adminUser');
+    }
+    //TODO: Fails with "Error: Unable to find an existing account matching your username for this team. This team may require an invite from the team owner to join."
+    // await cleanUpBotDMs(adminClient, adminUser.id, pluginId);
```

Comment:
@mickmister This line for some reason fails on my local machine. I'm not sure why. 
Do you have any thoughts?

Reply:
Fixed this. Looks like we had to pass the `botUsername` instead of pluginId for the 3rd parameter. For the GitHub plugin, the botUsername was same as the pluginID and hence it worked but in this case, both were different leading it to an error.

I've stored the `botUsername` in a separate variable and I'm using that instead. It works now!